### ISSUE TITLE
Look for advice for my Photo OCR application

### ISSUE THREAD

**Issue by bshao001** [experiencor/basic-yolo-keras#41]
I am working on a Photo OCR application. The objects I am trying to detect in an image are a few text areas. I am attaching one of the sample images here. (the text areas I am trying to detect are those dollar numbers. This output will then feed into models for segmentation, and lastly character recognition). The number of text area vary from 1 to 3. I don't know what is the most appropriate choice. The original image is 640 * 480. I can crop it into 480 * 480 as most likely, the text areas won't appear in the very left or very right.

1. Use grid. To me, it is really a waste of effort. Because the text area can never be more than 3. However, I will still have to use something like 1 * 12 grid. If that does not work, I may have to use 12 * 12 grid.  BTW, I saw the model is using 13 * 13 grid, which happens to be the size of the final layer. If that is a must, I will have to use 15 * 15 grid.

2. No grid. However, all the text areas will have almost the same shape. The anchor boxes seem not work for that.

Therefore, I would really appreciate if you could give me some advice as an experienced user on this model.

Thank you so much!

![sign1](https://user-images.githubusercontent.com/26190682/32757946-059863ce-c8b0-11e7-90bd-a3af4ff62fdd.jpg)

**Comment by experiencor** [comment]
@bshao001 

1. If you set input_size to 480, the resulting grid size is 15 x 15. 480/32 = 15. Setting input_size to (32, 384) will result in grid size of 1 x 12. It's worth a try.

2. For "all the text areas will have almost the same shape", do you mean sizes of bounding boxes are **uniform across all images**? If this is the case, you can just use one size for all anchors boxes. You still need to use multiple anchors box for each cell as many text areas may fall into the same cell. However, if you are sure that **each cell only contains one box**, you can just use one anchor box for one cell.

Hope it helps.

**Comment by bshao001** [comment]
@experiencor Thank you so much for your response.

As the size of the output layer has to match the grid size, it looks I have to use grid in order to use this model. Is that right? 

If we have the existing max pooling choices, the size of each cell in the grid is fixed to 32 * 32, is that right? 

It looks such a limitation to me. In the recent Andrew Ng's class, he is mentioning 19 * 19 grid, and each grid is very small. Was that also with cell size 32 * 32?

I am pretty sure that if the grid size is 15 * 15, then no two text areas (the center of the text areas) will fall into one cell, which has size 32 * 32. In this case, can I not to use anchor boxes at all? If it is possible, the loss function will be much simpler. Can you please give me some detailed guidance of how to modify your loss function for this?

Thank you so much again!

**Comment by experiencor** [comment]
It works by proportion. The ratio between input_size and grid_size is always 32 (true Tiny Yolo and Full Yolo). So if you want grid_size of 19 x 19, you must use input_size of 608, and so on. The ratio of 32 can also be changed by changing the layers of YOLO backends. I'm not recommending this as you will have to go through the trouble of training the backend, e.g. on COCO, again.

If no two text areas fall into one cell, you can just one anchor (so equivalent to no anchors at all). Don't have to modify the loss function. You should choose the size of the anchor intelligently. It should equal to the prevalent size of the bounding boxes. Note that the size of the anchor is relative to the size of the cell. For example, anchor of size [1, 1] means it's the size of the cell.

**Comment by bshao001** [comment]
Thank you so much. I will give it a try.
